pelicun/file_io.py

```diff
@@ -282,26 +286,53 @@ def substitute_default_path(
       are located.
     - If a path in the input list does not contain 'PelicunDefault/',
       it is added to the output list unchanged.
+    - If the file after 'PelicunDefault/' does not exist in the
+      `resource_paths` keys, a `KeyError` is raised.
 
     Examples
     --------
-    >>> data_paths = ['PelicunDefault/data/file1.txt',
-        'data/file2.txt']
+    >>> data_paths = ['PelicunDefault/data/file1.txt', 'data/file2.txt']
     >>> substitute_default_path(data_paths)
-    ['{base.pelicun_path}/resources/SimCenterDBDL/data/file1.txt',
+    ['{base.pelicun_path}/resources/SimCenterDBDL/updated_path/file1_v2.txt',
     'data/file2.txt']
 
     """
+    # Load the resource paths from the JSON file
+    resource_file_path = (
+        Path(base.pelicun_path) / 'resources' / 'dlml_resource_paths.json'
+    )
+    with resource_file_path.open('r') as file:
+        resource_paths = json.load(file)
+
     updated_paths: list[str | pd.DataFrame] = []
-    for data_path in data_paths:
-        if isinstance(data_path, str) and 'PelicunDefault/' in data_path:
-            path = data_path.replace(
-                'PelicunDefault/',
-                f'{base.pelicun_path}/resources/SimCenterDBDL/',
+    for data_path_str in data_paths:
+        if isinstance(data_path_str, str) and 'PelicunDefault/' in data_path_str:
+            data_path = Path(data_path_str)
+            # Extract the filename after 'PelicunDefault/'
+            file_name = (
+                data_path.parts[-1]
+                if 'PelicunDefault' in data_path.parts
+                else data_path.name
```

Comment:
Since we only enter this part if `'PelicunDefault/' in data_path_str`, I don't understand why we need to check if `'PelicunDefault' in data_path.parts`. The only way I can imagine this not to come back True is if the `data_path_str` is invalid and contains something like `"PelicunDefaultXYZ/..."` where XYZ is not supposed to be there. We could put an assertion after the Path() is created to test for this. If it fails, I'd just throw an error.
But, when would we ever use the `file_name = data_path.name`, that is, the `else` part of the above check?

Reply:
I agree that this was a redundant check., probably the result of not doing enough cleanup after arriving at a working prototype. The check in the enclosing `if` block suffices, and I don't think there is a strong need to further validate it.